src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -756,3 +757,30 @@ func (c *controller) UploadLogs(ctx context.Context, cancellog context.CancelFun
 		}
 	}
 }
+
+func (c controller) SetReadyState() {
+	c.log.Infof("Start waiting to be ready")
+	_ = utils.WaitForPredicate(WaitTimeout, 1*time.Second, func() bool {
+		_, err := c.ic.GetCluster(context.TODO())
+		if err != nil {
+			c.log.WithError(err).Warningf("Failed to connect to assisted service")
+			return false
+		}
+
+		_, err = c.kc.ListNodes()
+		if err != nil {
+			c.log.WithError(err).Warningf("Failed to connect to ocp cluster")
+			return false
+		}
+
+		c.log.Infof("I am ready, spawning event")
+		if _, err := c.kc.CreateEvent(c.Namespace, common.AssistedControllerIsReadyEvent,
+			"Assisted controller is ready", common.AssistedControllerPrefix); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
ready - ? connected to the assisted-service?

Reply:
It means it can talk to assisted-service and kube-apiserver

Reply:
Ready is the same as in any service, it means that i am ok and ready to run.

Reply:
I know. Will anyone else know what it means? Lets add it to the message?

Reply:
sure